### PR TITLE
Add external/SSO authentication for CLI

### PR DESCRIPTION
**Pull request by mosabua**
Replaces https://github.com/trinodb/trino/pull/7335

### CODE REVIEW

File: docs/src/main/sphinx/installation/cli.rst [hunk 1]
```diff
@@ -58,6 +58,29 @@ password value to avoid the prompt.
...
+
+* Start the CLI with the ``--external-authentication`` option.
+* CLI starts and connects to Trino.
+* A message appears in the CLI directing you to open a browser with a specified
+  URL when the first query is submitted.
+* Open the URL in a browser and follow through the authentication process.
+* The CLI automatically receives a token.
+* When successfully authenticated in the browser, the CLI proceeds to execute
+  the query.
+* Further queries in the CLI session do not require additional logins, while the
```
Comment (m57lyra) [inline docs/src/main/sphinx/installation/cli.rst:79]: remove the comma before while

**Review by m57lyra — APPROVED** [review]
LGTM

File: docs/src/main/sphinx/installation/cli.rst [hunk 2]
```diff
@@ -58,6 +58,29 @@ password value to avoid the prompt.
 Use ``--help`` to see information about specifying the keystore, truststore, and
 other authentication details as required. If using Kerberos, see :doc:`/security/cli`.
 
+External authentication - SSO
+-----------------------------
+
+The  ``--external-authentication`` option is used for browser-based SSO
+authentication, as detailed in :doc:`/security/oauth2`. With this configuration,
+the CLI displays a URL that you have to open in a web browser for
```
Comment (Ordinant) [inline docs/src/main/sphinx/installation/cli.rst:66]: s/have to/must

File: docs/src/main/sphinx/installation/cli.rst (hunk 1 above)
Comment (Ordinant) [reply docs/src/main/sphinx/installation/cli.rst:79]: > remove the comma before while

What she said

File: docs/src/main/sphinx/installation/cli.rst [hunk 3]
```diff
@@ -58,6 +58,29 @@ password value to avoid the prompt.
...
+* Start the CLI with the ``--external-authentication`` option.
+* CLI starts and connects to Trino.
+* A message appears in the CLI directing you to open a browser with a specified
+  URL when the first query is submitted.
+* Open the URL in a browser and follow through the authentication process.
+* The CLI automatically receives a token.
+* When successfully authenticated in the browser, the CLI proceeds to execute
+  the query.
+* Further queries in the CLI session do not require additional logins, while the
+  authentication token remains valid. Token expiry depends on the external
```
Comment (Ordinant) [inline docs/src/main/sphinx/installation/cli.rst:80]: Token expiration depends on ...

**Review by Ordinant — CHANGES_REQUESTED** [review]
Comments interspersed.

File: docs/src/main/sphinx/installation/cli.rst (hunk 1 above)
Comment (mosabua) [reply docs/src/main/sphinx/installation/cli.rst:79]: Done

File: docs/src/main/sphinx/installation/cli.rst (hunk 2 above)
Comment (mosabua) [reply docs/src/main/sphinx/installation/cli.rst:66]: done

File: docs/src/main/sphinx/installation/cli.rst (hunk 3 above)
Comment (mosabua) [reply docs/src/main/sphinx/installation/cli.rst:80]: done

File: docs/src/main/sphinx/installation/cli.rst [hunk 4]
```diff
@@ -58,6 +58,28 @@ password value to avoid the prompt.
 Use ``--help`` to see information about specifying the keystore, truststore, and
 other authentication details as required. If using Kerberos, see :doc:`/security/cli`.
 
+External authentication - SSO
+-----------------------------
+
+The  ``--external-authentication`` option is used for browser-based SSO
```
Comment (kokosing) [inline docs/src/main/sphinx/installation/cli.rst:64]: nit: double space after `The`

**Review by kokosing — APPROVED** [review]
@s2lomon Please take a look.

**Review by s2lomon — APPROVED** [review]
lgtm

File: docs/src/main/sphinx/installation/cli.rst (hunk 4 above)
Comment (mosabua) [reply docs/src/main/sphinx/installation/cli.rst:64]: Fixed

**Comment by mosabua** [comment]
Can we get this merged please @electrum?

**Pull request merged**